craft_parts/executor/step_handler.py

```diff
@@ -288,32 +293,63 @@ def _handle_control_api(self, scriptlet_name: str, function_call: str) -> None:
                     f"{scriptlet_name!r} control call missing attribute {attr!r}"
                 )
 
-        function_name = function_json["function"]
-        function_args = function_json["args"]
+        cmd_name = function_json["function"]
+        cmd_args = function_json["args"]
+
+        return self._process_api_commands(
+            cmd_name, cmd_args, step=step, scriptlet_name=scriptlet_name
+        )
+
+    def _process_api_commands(
+        self, cmd_name: str, cmd_args: List[str], *, step: Step, scriptlet_name: str
+    ) -> str:
+        """Invoke API command actions."""
+        retval = ""
 
         invalid_control_api_call = functools.partial(
             errors.InvalidControlAPICall,
             part_name=self._part.name,
             scriptlet_name=scriptlet_name,
         )
 
-        if function_name in ["pull", "build", "stage", "prime"]:
-            if len(function_args) > 0:
+        if cmd_name == "default":
```

Comment:
we might be missing explanation documentation attached to this this

Reply:
or about craftctl in general

Reply:
Will add documentation in a separate PR, creating a task for this.